lib/ZwaveClient.ts

```diff
@@ -1020,48 +1022,18 @@ class ZwaveClient extends TypedEventEmitter<ZwaveClientEventCallbacks> {
 					zwaveNode
 				),
 		}
+
 		const weeklySchedules: (ScheduleEntryLockWeekDaySchedule &
 			ScheduleEntryLockSlotId)[] = []
 		const yearlySchedules: (ScheduleEntryLockYearDaySchedule &
 			ScheduleEntryLockSlotId)[] = []
 		const dailySchedules: (ScheduleEntryLockDailyRepeatingSchedule &
 			ScheduleEntryLockSlotId)[] = []
-		for (let i = 1; i <= userCodes; i++) {
-			for (let s = 1; s <= numSlots.numWeekDaySlots; s++) {
-				const slot: ScheduleEntryLockSlotId = {
-					userId: i,
-					slotId: s,
-				}
-				const schedule = await zwaveNode.commandClasses[
-					'Schedule Entry Lock'
-				].getWeekDaySchedule(slot)
-				if (schedule) weeklySchedules.push({ ...slot, ...schedule })
-			}
 
-			for (let s = 1; s <= numSlots.numYearDaySlots; s++) {
-				const slot: ScheduleEntryLockSlotId = {
-					userId: i,
-					slotId: s,
-				}
-				const schedule = await zwaveNode.commandClasses[
-					'Schedule Entry Lock'
-				].getYearDaySchedule(slot)
-				if (schedule) yearlySchedules.push({ ...slot, ...schedule })
-			}
+		const node = this._nodes.get(nodeId)
 
-			for (let s = 1; s <= numSlots.numDailyRepeatingSlots; s++) {
-				const slot: ScheduleEntryLockSlotId = {
-					userId: i,
-					slotId: s,
-				}
-				const schedule = await zwaveNode.commandClasses[
-					'Schedule Entry Lock'
-				].getDailyRepeatingSchedule(slot)
-				if (schedule) dailySchedules.push({ ...slot, ...schedule })
-			}
-		}
+		const isInited = !!node.schedule
 
-		const node = this._nodes.get(nodeId)
 		node.schedule = {
```

Comment:
I don't understand this tbh. You check first if node.schedule is defined, then you override it.

Reply:
Also, are the schedules cached locally or do you re-query every time this is called?

Reply:
The diff doens't help understanding this change, you should see the end result. BTW I firstly check if schedule is defined because I call `getSchedule` on node ready, the first time is called so it just create the `node.schedule` object with the cached info, on next calls it also query the slots. So that check is used to allow init node.schedule and then skip the query that is done in the `for` loop.

Reply:
> Also, are the schedules cached locally or do you re-query every time this is called?

Them are cached on node, not on disk BTW